Source/DafnyCore/Resolver/RunAllTestsMainMethod.cs

```diff
@@ -19,17 +25,12 @@ public class RunAllTestsMainMethod : IRewriter {
   /// errors much earlier in the pipeline.
   /// </summary>
   internal override void PreResolve(Program program) {
-    var hasMain = Compilers.SinglePassCompiler.HasMain(program, out var mainMethod);
-    if (hasMain) {
-      Reporter.Error(MessageSource.Rewriter, mainMethod.tok, "Cannot use /runAllTests on a program with a main method");
-      return;
-    }
```

Comment:
The code at the end of this file still calls `hasMain` and an assertion fails when running with `/testContracts`. I wonder whether the implementation of `HasMain` need to be updated to account for the possibility of the new synthetic name you're using?

Reply:
That was a satisfyingly simple fix. :)